### PR TITLE
Feature/add transactions to homepage

### PR DESCRIPTION
**Pull request by karthikeyan-cere**
1. Added trigger to updated signed_transactions for every extrinsic
2. Updated home page to show signed_transactions
image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Cerestats
 
 ## vNext
-- ...
+- Added signed transaction count to homepage.
```
Comment (AndreiNavoichyk) [inline CHANGELOG.md]: transaction**s**

File: backend/docker/sql/backend-dev.sql [hunk 2]
```diff
@@ -235,6 +235,7 @@ INSERT INTO total (name, count) VALUES
   ('nominator_count', 0),
   ('current_era', 0),
   ('active_era', 0),
+  ('signed_transactions', 0),
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 3]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
 -- initialize the counter table
 UPDATE total SET count = (SELECT count(*) FROM extrinsic WHERE method IN ('transfer', 'transferKeepAlive')) WHERE name = 'transfers';
 COMMIT;
+
+-- Signed Transactions
+START TRANSACTION;
+CREATE FUNCTION signed_transaction_count() RETURNS trigger LANGUAGE plpgsql AS
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 4]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
...
+$$BEGIN
+  IF NEW.is_signed = true THEN
+    IF TG_OP = 'INSERT' THEN
+      UPDATE total SET count = count + 1 WHERE name = 'signed_transactions';
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 5]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
...
+      RETURN NEW;
+    ELSIF TG_OP = 'DELETE' THEN
+      UPDATE total SET count = count - 1 WHERE name = 'signed_transactions';
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 6]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
...
+      RETURN OLD;
+    ELSE
+      UPDATE total SET count = 0 WHERE name = 'signed_transactions';
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 7]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
...
+      RETURN NULL;
+    END IF;
+  END IF;
+  RETURN NULL;
+END;$$;
+CREATE CONSTRAINT TRIGGER signed_transaction_count_mod
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 8]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
...
+  AFTER INSERT OR DELETE ON extrinsic
+  DEFERRABLE INITIALLY DEFERRED
+  FOR EACH ROW EXECUTE PROCEDURE signed_transaction_count();
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 9]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
...
+-- TRUNCATE triggers must be FOR EACH STATEMENT
+CREATE TRIGGER signed_transaction_count_trunc AFTER TRUNCATE ON extrinsic
+  FOR EACH STATEMENT EXECUTE PROCEDURE signed_transaction_count();
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: backend/docker/sql/backend-dev.sql [hunk 10]
```diff
@@ -384,3 +385,32 @@ CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
...
+-- initialize the counter table
+UPDATE total SET count = (SELECT count(*) FROM extrinsic WHERE is_signed = true) WHERE name = 'signed_transactions';
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql]: signed_extrinsics

File: frontend/components/Chain.vue [hunk 11]
```diff
@@ -256,8 +262,9 @@ export default {
         result({ data }) {
           this.totalExtrinsics =
             data.total.find((row) => row.name === 'extrinsics').count || 0
-          this.totalTransfers =
-            data.total.find((row) => row.name === 'transfers').count || 0
+          this.totalSignedTransactions =
```
Comment (AndreiNavoichyk) [inline frontend/components/Chain.vue]: totalSignedExtrinsics

**Review by AndreiNavoichyk — CHANGES_REQUESTED** [review]
Substrate does not have transactions. It has extrinsics. You should use `extrinsics` inside the whole data and business layer. Please, rename.

File: CHANGELOG.md (hunk 1 above)
Comment (karthikeyan-cere) [reply CHANGELOG.md]: Done

File: backend/docker/sql/backend-dev.sql (hunk 2 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 3 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 4 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 5 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 6 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 7 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 8 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 9 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: backend/docker/sql/backend-dev.sql (hunk 10 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql]: Done

File: frontend/components/Chain.vue (hunk 11 above)
Comment (karthikeyan-cere) [reply frontend/components/Chain.vue]: Done

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,7 +1,7 @@
 # Cerestats
 
 ## vNext
-- ...
+- Added signed transactions count to homepage.
```
Comment (AndreiNavoichyk) [inline CHANGELOG.md]: Rename to total transactions here

**Review by AndreiNavoichyk — APPROVED** [review]

File: CHANGELOG.md (hunk 12 above)
Comment (karthikeyan-cere) [reply CHANGELOG.md]: Updated.

**Pull request merged**